src/goto-symex/builtin_functions.cpp

```diff
@@ -1673,6 +1673,367 @@ static inline expr2tc gen_value_by_byte(
   return gen_byte_expression(type, src, value, num_of_bytes, offset);
 }
 
+// Computes the equivalent object value when considering a memcpy operation on it
+static inline expr2tc gen_byte_memcpy(
```

Comment:
Shall we create unit tests for this method? The logic itself is intricate and might be prone to off-by-one or bit-shifting errors :-)